evm/executeContract.js

```diff
@@ -0,0 +1,244 @@
+        tokenValue = Number(0);
+    }
+
+    (async () => {
+        try {
+            const result = await contract.executeContract(targetContractAddress, callData, tokenValue);
+            printLog('Function successfully called with return value as: ');
+            printObj(result);
+        } catch (error) {
+            printError('Calling executeContract method failed with Error: ', error);
+        }
+    })();
+}
+
+async function main(options) {
+    const config = loadConfig(options.env);
+    let chains = options.chainNames.split(',').map((str) => str.trim());
+
+    if (options.chainNames === 'all') {
+        chains = Object.keys(config.chains);
+    }
+
+    for (const chain of chains) {
+        if (config.chains[chain.toLowerCase()] === undefined) {
+            throw new Error(`Chain ${chain} is not defined in the info file`);
+        }
+    }
+
+    for (const chain of chains) {
+        const rpc = config.chains[chain.toLowerCase()].rpc;
+        const provider = getDefaultProvider(rpc);
+        const privateKey = options.privateKey;
+
+        if (!isString(privateKey)) {
+            throw new Error('Private Key value is not provided in the info file');
+        }
+
+        const wallet = new Wallet(privateKey, provider);
+        await executeContract(
+            options.pathCallContract,
+            options.pathTargetContract,
+            options.callContractAddress,
+            options.targetAddress,
+            wallet,
+            options.callData,
+            Number(options.nativeValue),
+            options.methodName,
+            options.recipientAddress,
+            Number(options.amount),
+        );
+        saveConfig(config, options.env);
+    }
+}
+
+const program = new Command();
+
+program.name('execute-contract').description('Executes a call to an external contract');
+
+program.addOption(
+    new Option('-e, --env <env>', 'environment')
+        .choices(['local', 'devnet', 'stagenet', 'testnet', 'mainnet'])
+        .default('testnet')
+        .makeOptionMandatory(true)
+        .env('ENV'),
+);
+program.addOption(new Option('-n, --chainNames <chainNames>', 'chain names').makeOptionMandatory(true));
+program.addOption(
+    new Option('-pc, --pathCallContract <pathCallContract>', 'artifact path for the called contract').makeOptionMandatory(true),
+);
+program.addOption(
+    new Option('-pt, --pathTargetContract <pathTargetContract>', 'artifact path for the target contract').makeOptionMandatory(true),
+);
+program.addOption(
+    new Option('-a, --callContractAddress <contractAddress>', 'The contract address in which we will call executeContract function')
+        .makeOptionMandatory(true)
+        .env('CONTRACT_ADDR'),
+);
+program.addOption(
+    new Option('-t, --targetAddress <targetAddress>', 'The address of the contract to be called')
+        .makeOptionMandatory(true)
+        .env('TARGET_ADDR'),
```

Comment:
Let's discuss with @milapsheth but we might always want to pass target address, native value, and method name via the command line

Reply:
Removing native value and method name from env. For the target address its your call, we can discuss with @milapsheth as well.

Reply:
@milapsheth can you pease comment here.

Reply:
yeah, cli works